test/framework/clusterctl/clusterctl_helpers.go

```diff
@@ -327,6 +328,11 @@ func ApplyClusterTemplateAndWait(ctx context.Context, input ApplyClusterTemplate
 		Lister:  input.ClusterProxy.GetClient(),
 		Cluster: result.Cluster,
 	}, input.WaitForMachinePools...)
+
+	if input.WaitForMachinesReady != nil {
```

Comment:
It feels a little odd having this as an optional waiter (I think all  of the other Waiters here are compulsory). Could you instead make this a input.PostMachinesProvisioned() function, so it fits in with e.g. PreWaitForCluster on L:282?

Reply:
I didn't notice that pattern, yeah, good idea.

Reply:
It's just merged in the last few days :slightly_smiling_face: